From 66286e77b663f60cea152fc96bdd0af2ad7dbd68 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 03 Jul 2023 13:06:59 +0300
Subject: [PATCH] [SW-150757] arcboot: use position as a number instead of enum

both arcboot_rom_status_update_bringup() and security_load_keys() get
2 arguments, a progress an position (loc). since the position can be
stored as different enums, the argument type changed from a specific
enum type to an unsigned int.

Change-Id: Iec56f89244ed5daafb297445664e00643401d0b1
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 0340f12..1861acb 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -45,14 +45,12 @@
 	regmap_writel(reg, (mmSW_STATUS_0 + offset * 4));
 }
 
-void arcboot_rom_status_update_bringup(enum arcboot_rom_progress progress,
-					      enum arcboot_rom_bringup_proggress position)
+void arcboot_rom_status_update_bringup(enum arcboot_rom_progress progress, uint32_t position)
 {
 	arcboot_rom_status_update(progress + position);
 }
 
-void arcboot_rom_status_update_location(uint8_t progress,
-					enum security_location loc)
+void arcboot_rom_status_update_location(uint8_t progress, uint32_t loc)
 {
 	uint32_t reg;
 
diff --git a/include/arcboot.h b/include/arcboot.h
index e6fb0e6..8e758ed 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -183,10 +183,8 @@
 extern struct arcboot_global_s arcboot_rom_global;
 extern struct arcboot_manifest_s arcboot_rom_manifest;
 void arcboot_rom_status_update(enum arcboot_rom_progress progress);
-void arcboot_rom_status_update_bringup(enum arcboot_rom_progress progress,
-					enum arcboot_rom_bringup_proggress position);
-void arcboot_rom_status_update_location(uint8_t progress,
-					enum security_location loc);
+void arcboot_rom_status_update_bringup(enum arcboot_rom_progress progress, uint32_t position);
+void arcboot_rom_status_update_location(uint8_t progress, uint32_t loc);
 void arcboot_rom_bringup_ppboot(void);
 void arcboot_rom_validation_fail(void);
 void arcboot_rom_clear_recovery_flag(void);
